docs/userguide/modules/creating-service-bindings/pages/custom-path-injection.adoc

```diff
@@ -132,3 +132,77 @@ metadata:
+metadata:
+  name: # string <1>
+  generation: # int64, defined by the Kubernetes control plane
+  ...
+spec:
+  versions: # required, at least one entry must be defined
+  - version: # string <2>
+    annotations: # string (Fixed JSONPath), optional <3>
+    containers: # optional
+    - path: # string (JSONPath) <4>
+      name: # string (Fixed JSONPath), optional <5>
+      env: # string (Fixed JSONPath), optional <6>
+      volumeMounts: # string (Fixed JSONPath), optional <7>
+    volumes: # string (Fixed JSONPath), optional <8>
+....
+<1> the name of this resource (must be qualified as `plural.group`, i.e. `cronjobs.batch` for
+CronJobs)
+<2> the version of the resource we're mapping (can match all unspecified versions with the wildcard "*")
+<3> where annotations live with a fixed JSONPath (default: `.spec.template.spec.annotations`)
+<4> where containers live with a JSONPath (if no entries are defined, SBO defaults to two paths:
+`.spec.template.spec.containers` and `.spec.template.spec.initContainers` with all other fields
+set as their default).
+<5> corresponds to the `.name` field in a pod. Defaults to `.name`.
+<6> corresponds to the `.env` environment variables. Defaults to `.env`.
+<7> corresponds to where the volume mounts of a pod lives. Defaults to `.volumeMounts`.
+<8> where volumes live with a fixed JSONPath (default: `.spec.template.spec.volumes`)
+
+In this context, a "Fixed JSONPath" is a subset of the JSONPath grammar that accepts only two
+operations: field lookup (i.e. `.spec.template`) and indexing (i.e. `.spec['template']`).  All other
+operations are forbidden.
+
+Most of these fields are optional, and SBO assumes PodSpec-compatible defaults when not specified.
+SBO requires that each of these fields is semantically equivalent to the corresponding field in a Pod
+deployment; failing to maintain this may result in unexpected behavior.
+
+Using this schema, we can write an example mapping for `CronJob.batch/v1` resources:
+
+[source,yaml]
+....
+apiVersion: servicebinding.io/v1alpha3
+kind: ClusterWorkloadResourceMapping
+metadata:
+ name: cronjobs.batch
+spec:
+  versions:
+  - version: "v1"
+    annotations: .spec.jobTemplate.spec.template.metadata.annotations
+    containers:
+    - path: .spec.jobTemplate.spec.template.spec.containers[*]
+    - path: .spec.jobTemplate.spec.template.spec.initContainers[*]
+    volumes: .spec.jobTemplate.spec.template.spec.volumes
+....
+
+NOTE: There is some behavior in the CoreOS API that doesn't exist in the specification API:
+
+. If a service binding with `bindAsFiles: false` is created with one of these mappings, then
+environment variables will be projected into `.envFrom` underneath each `path` field specified in the
+corresponding `ClusterWorkloadResourceMapping` resource.
+. Both this and the `containersPath` field documented above can be used at the same time.  This would
+be equivalent to adding a container entry with `path: $(containersPath)`.
```

Comment:
```suggestion
[NOTE]
====
The following behavior occurs only in the `binding.operators.coreos.com` API group:

* If a service binding with the `bindAsFiles: false` flag value is created with one of these mappings, then environment variables are projected into the `.envFrom` field under each `path` field specified in the corresponding `ClusterWorkloadResourceMapping` resource.
* Both the fields `.envFrom` and `containersPath` can be in use at the same time. This is equivalent to adding a container entry with the `path: $(containersPath)` attribute.
```
Question:
- In the sentence at L207: "Both this ..", What do you mean exactly by "this"? Please clarify and provide inputs.
- Again, in "...can be used at the same time", What do you mean exactly? Used for what, who is using? Please clarify and provide inputs.
- The 2nd bullet item is very confusing and difficult to comprehend. Please rewrite for clarity.

Reply:
> In the sentence at L207: "Both this ..", What do you mean exactly by "this"? Please clarify and provide inputs.

By "this", I mean `ClusterWorkloadResourceMapping` resources.

> Again, in "...can be used at the same time", What do you mean exactly? Used for what, who is using? Please clarify and provide inputs.

It is possible for a cluster administrator to specify both a `ClusterWorkloadResourceMapping` resource and the `.spec.application.bindingPath.containersPath` field in a `ServiceBinding` resource.  SBO will attempt to project binding information into the locations specified in both of these methods.  That is the behavior I was attempting to document.

> The 2nd bullet item is very confusing and difficult to comprehend. Please rewrite for clarity.

I'll take another attempt at it - I agree it's not the clearest.

Reply:
I've taken another attempt at writing this bullet point, PTAL.